xmppserver/src/main/java/org/jivesoftware/openfire/session/LocalSession.java

```diff
@@ -434,44 +451,64 @@ public void deliverRawText(String text) {
 
     @Override
     public void close() {
-        if (conn == null) return;
-        conn.close();
+        lock.lock();
+        try {
+            Optional.ofNullable(conn)
+              .ifPresent(connection ->  connection.close());
+        }finally {
+            lock.unlock();
+        }
     }
 
     @Override
     public boolean validate() {
-        return conn.validate();
+        lock.lock();
```

Comment:
Don't think this lock/unlock is necessary

Reply:
done